src/core/a-entity.js

```diff
@@ -73,19 +73,21 @@ var proto = Object.create(ANode.prototype, {
       // Don't .load() scene on attachedCallback.
       if (this.isScene) { return; }
 
-      // Gracefully not error when outside of <a-scene> (e.g., tests).
-      if (!sceneEl) {
-        this.load();
-        return;
-      }
+      utils.waitForDOMContentLoaded().then(function () {
```

Comment:
What about just deferring in like 70 do:

```utils.waitForDOMContentLoaded(this.attachedCallback.bind(this));```

it should solve for both scene and entities I believe.

Reply:
Even better would be to do it in the attachedCallback for a-entity.js

Reply:
I actually tried to put that in a-node attachedCallback first, but some tests failed.
But actually I don't think it makes sense, see a-scene attachedCallback is calling a-entity attachedCallback which is calling a-node attachedCallback. Putting a Promise in a-node attachedCallback means the caller a-entity attachedCallback may not have some attributes set yet like this.sceneEl. The promise needs to be at the top of the calling chain, so this probably make more sense to do patch attachedCallback to do thhe waitForDOMContentLoaded in registerElement after we do the wrapMethods.
This is what I did in the latest commit, I patched the attachedCallback for any registered elements that uses AEntity prototype (so a-scene, a-entity and all primitives a-camera a-gltf-model etc).
I didn't do it for registered elements that uses the ANode prototype (a-assets, a-mixin), some a-assets tests are failing if I do that, but there is not really an issue of system or component init in those anyway.